sql/catalyst/src/main/scala/org/apache/spark/sql/types/Decimal.scala

```diff
@@ -134,7 +135,14 @@ final class Decimal extends Ordered[Decimal] with Serializable {
   def set(bigintval: BigInteger): Decimal = {
     try {
       this.decimalVal = null
-      this.longVal = bigintval.longValueExact()
+      // TODO: Remove this once we migrate to java8 and use longValueExact() instead.
+      val isExactLong = bigintval.compareTo(BigInteger.valueOf(JLong.MAX_VALUE)) <= 0 &&
```

Comment:
Maybe extract constants for a `BigInteger` representation of the max/min values rather than compute each time? Also is this all uses of java.util.Long that can be JLong?